contracts/modules/Checkpoint/ScheduledCheckpoint.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity ^0.4.24;
+
+import "./ICheckpoint.sol";
+import "../TransferManager/ITransferManager.sol";
+import "../../interfaces/ISecurityToken.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Burn module for burning tokens and keeping track of burnt amounts
+ */
+contract ScheduledCheckpoint is ICheckpoint, ITransferManager {
+    using SafeMath for uint256;
+
+    struct Schedule {
+        bytes32 name;
+        uint256 startTime;
+        uint256 nextTime;
+        uint256 interval;
+        uint256 index;
+        uint256[] checkpointIds;
+        uint256[] timestamps;
+        uint256[] periods;
+    }
+
```

Comment:
one more thing we can do is compact binding in the struct if want to save some gas for the transaction.